### PR TITLE
League statistics

### PR DESCRIPTION
**Pull request by kevxo**
Creates League statistics class with test. There is also refactoring involved with the class itself and game statistics test.
I added also all the methods we've done so far from the game and league stats to startrack class to make spec harness pass.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -24,7 +27,7 @@ def game_stats
   end
 
   def game_teams_stats
-    game_teams_data = CSV.read(@game_teams, {encoding: "UTF-8", headers: true, header_converters: :symbol, converters: :all})
+    game_teams_data = CSV.read(@game_teams, { encoding: 'UTF-8', headers: true, header_converters: :symbol, converters: :all })
```
Comment (Oxalisviolacea) [inline lib/stat_tracker.rb:30]: We should see if this works without the encoding: "UTF-8" argument.

File: test/league_stats_test.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    locations = {
+      games: game_path,
+      teams: team_path,
+      game_teams: game_teams_path
+    }
+
+    @stat_tracker = StatTracker.from_csv(locations)
+    @raw_teams_stats = @stat_tracker.teams_stats
+    @raw_game_stats = @stat_tracker.game_stats
```
Comment (Oxalisviolacea) [inline test/league_stats_test.rb:20]: Do these need to be instance variables?

**Review by Oxalisviolacea — APPROVED** [review]

File: test/league_stats_test.rb (hunk 2 above)
Comment (kevxo) [reply test/league_stats_test.rb:20]: Yes they are part of the test set-up

File: test/game_statistics_test.rb [hunk 3]
```diff
@@ -14,10 +14,10 @@ def setup
       game_teams: game_teams_path
     }
 
-    @stat_tracker_location = StatTracker.from_csv(locations)
-    @stat_tracker = @stat_tracker_location.game_stats
-    @stat_game_teams_tracker = @stat_tracker_location.game_teams_stats
-    @game_statistics = GameStatistics.new(@stat_tracker, @stat_game_teams_tracker)
+    @stat_tracker = StatTracker.from_csv(locations)
+    @raw_game_stats = @stat_tracker.game_stats
```
Comment (ShaunDaneJames) [inline test/game_statistics_test.rb:18]: I REALLY appreciate the naming here, it clears things up.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -35,4 +38,78 @@ def game_teams_stats
...
+
+  def count_of_teams
+    @league_statistics.count_of_teams
+  end
+
+  def best_offense
+    @league_statistics.best_offense_stats
+    @league_statistics.best_offense
+  end
+
```
Comment (ShaunDaneJames) [inline lib/stat_tracker.rb:90]: Are these methods an opportunity for modules?

**Review by ShaunDaneJames — APPROVED** [review]

File: lib/game_statistics.rb [hunk 5]
```diff
@@ -76,11 +76,11 @@ def average_goals_per_game
 
   def average_goals_by_season
     hash = {}
-    hash_of_seasons.each do |key, value|
-      x = value.map do |game|
+    hash_of_seasons.each do |season, stat|
+      goals_per_szn = stat.map do |game|
```
Comment (carson-jardine) [inline lib/game_statistics.rb:80]: like the name changes!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (carson-jardine) [reply lib/stat_tracker.rb:30]: 2nd that!

File: lib/stat_tracker.rb (hunk 4 above)
Comment (carson-jardine) [reply lib/stat_tracker.rb:90]: I think they absolutely are, we should look into that when refactoring!

File: test/league_stats_test.rb [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  def test_count_of_teams
+    assert_equal 32, @league_stats.count_of_teams
+  end
+
+  def test_group_by_team_id
+    assert_equal 32, @league_stats.group_by_team_id.keys.count
+  end
+
+  def test_team_id_and_average_goals
+    assert_equal 32, @league_stats.team_id_and_average_goals.count
```
Comment (carson-jardine) [inline test/league_stats_test.rb:43]: These tests look like a great place to use mocks and stubs! Should we look into doing that when refactoring?

**Review by carson-jardine — APPROVED** [review]

**Pull request merged**